### PR TITLE
Move Linux Crossbuild definition to Docker-exec model

### PR DESCRIPTION
**Pull request by wtgodbe**
This changes the Dotnet-CoreFx-Trusted-Linux-Crossbuild job so that it uses the Docker exec model - I've tested this in VSO & it should work. Don't merge until Monday.

### CODE REVIEW

**Comment by wtgodbe** [comment]
@MattGal @chcosta @gkhanna79 PTAL

File: buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json [hunk 1]
```diff
@@ -481,61 +315,28 @@
...
       "inputs": {
-        "filename": "docker",
-        "arguments": "rm $(PB_DockerContainerName)",
-        "workingFolder": "",
-        "failOnStandardError": "false"
+        "CopyRoot": "$(PB_DockerCopyDest)/corefx",
+        "Contents": "**/*.log",
+        "ArtifactName": "BuildLogs",
+        "ArtifactType": "Container",
+        "TargetPath": "\\\\my\\share\\$(Build.DefinitionName)\\$(Build.BuildNumber)"
```
Comment (gkhanna79) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: Is this a well known path that we want to publish?

Comment (wtgodbe) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: This is the default path for publishing artifacts - see https://github.com/dotnet/corefx/blob/master/buildpipeline/DotNet-CoreFx-Trusted-Linux.json#L339. In fact, it doesn't even show up in the VSO UI.

**Comment by wtgodbe** [comment]
@MattGal @chcosta any issues with this PR? I'd like to get this merged today

File: buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json [hunk 2]
```diff
@@ -4,52 +4,32 @@
...
+      "displayName": "Delete files from $(Build.SourcesDirectory)",
       "timeoutInMinutes": 0,
       "task": {
-        "id": "d9bafed4-0b18-4f58-968d-86655b4d2ce9",
+        "id": "b7e8b412-0437-4065-9371-edc5881de25b",
         "versionSpec": "1.*",
         "definitionType": "task"
       },
       "inputs": {
-        "filename": "$(Build.SourcesDirectory)/Tools/scripts/docker/cleanup-docker.sh",
```
Comment (chcosta) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: we should leave the docker cleanup stuff if it's not causing a problem

File: buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json [hunk 3]
```diff
@@ -94,16 +74,16 @@
       "enabled": true,
       "continueOnError": false,
       "alwaysRun": false,
-      "displayName": "Run clean.sh",
+      "displayName": "Docker cleanup",
```
Comment (wtgodbe) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: @chcosta docker cleanup happens here

File: buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json (hunk 2 above)
Comment (chcosta) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: Oh, maybe this just got moved and looks weird in the diff..

File: buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json [hunk 4]
```diff
@@ -319,43 +191,7 @@
...
-      "timeoutInMinutes": 0,
-      "task": {
-        "id": "d9bafed4-0b18-4f58-968d-86655b4d2ce9",
-        "versionSpec": "1.*",
-        "definitionType": "task"
-      },
-      "inputs": {
-        "filename": "docker",
-        "arguments": "rm $(PB_DockerContainerName)",
+        "arguments": "docker exec --privileged $(PB_DockerContainerName) $(PB_GitDirectory)/cross/build-rootfs.sh $(PB_Architecture) $(CrossToolsetVersion) $(SkipUnmount)",
```
Comment (chcosta) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: `$(CrossToolsetVersion)` and `$(SkipUnmount)` come from pipebuild?  you may want to prefix them with PB_ to ensure there are no variable name conflicts between what pipebuild passes and what is explicitly being passed to the 'build-rootfs.sh' script.

Comment (chcosta) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: meh, I guess it's just a drop in variable, and shouldn't be a conflict.  it's just a little inconsistent

**Comment by MattGal** [comment]
LGTM

**Pull request merged**